internal/auth/authController.go

```diff
@@ -0,0 +1,72 @@
+package auth
+
+import (
+	"eventapi/internal/database"
+	"eventapi/internal/session"
+	"fmt"
+
+	"github.com/gofiber/fiber/v2"
+)
+
+const userInfoURL = "https://www.googleapis.com/oauth2/v2/userinfo"
+
+// Login handler
+func Login(a *AuthService) func(c *fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+		url := a.getEntryPoint()
+		return c.Redirect(url)
+	}
+
+}
+
+// oauth callback
+func Callback(oa oAuthService, u database.UserRepository, s *session.SessionService) func(c *fiber.Ctx) error {
+	return func(c *fiber.Ctx) error {
+		token, err := oa.exchange(c.Context(), c.FormValue("state"), c.FormValue("code"))
+		if err != nil {
+			return err
+		}
+
+		ui, err := getUserInfo(token.AccessToken)
+
+		if err != nil {
+			return fmt.Errorf("could not fetch user info: %w", err)
+		}
+
+		if exists := u.Exists(ui.ID); !exists {
+			println("User does not exist, creating")
+
+			_, err := u.Create(database.NewUser{
+				ID:           ui.ID,
+				FirstName:    ui.FirstName,
+				LastName:     ui.LastName,
+				Email:        ui.Email,
+				Picture:      ui.Picture,
+				Verified:     ui.Verified,
+				RefreshToken: token.RefreshToken,
+			})
```

Comment:
This is more of a business logic decision - whether you want anybody to be able to create an account (who has a valid Google user) or if you want to restrict it to be invite only. If invite only, you'd want to populate the DB with a whitelist of possible users (via email) who can create/use accounts.

Reply:
Also I keep saying email address mostly because i'm not familiar with what this ID is.

Reply:
I hadn't thought of how I was going to handle user management thanks for bringing this up. I am going to go with that instead of trying to manage through google beta access or something else as it just makes more sense.